contracts/base/LockAndMsgSender.sol

```diff
@@ -3,15 +3,29 @@ pragma solidity ^0.8.17;
 
 import {Constants} from '../libraries/Constants.sol';
 
-/// @title Recipient Library
-/// @notice Calculates the recipient address for a command
-library Recipient {
+contract LockAndMsgSender {
+    error ContractLocked();
+
+    address private constant ADDRESS_ONE = address(1);
+    address internal isLockedSender = ADDRESS_ONE;
+
+    modifier isNotLocked() {
+        if (msg.sender != address(this)) {
+            if (isLockedSender != ADDRESS_ONE) revert ContractLocked();
+            isLockedSender = msg.sender;
+            _;
+            isLockedSender = ADDRESS_ONE;
+        } else {
```

Comment:
Yes all reentrancy with `msg.sender==address(this)` would be allowed with this. However:

- only one function has the reentrancy lock so its only 1 entry point to consider
- all of the `.call(` operations in the codebase are performed upon a fixed address - mostly protocols but in 1 place it is `(address(this)).call`. This is _only_ within the `EXECUTE_SUB_PLAN` command which is defined in this PR.

So yes it does loosen the reentrancy lock a bit but its such a tight set of circumstances that it feels safe to me.

However, interested to know any concerns you have!

Reply:
Yah the only wayy ever this can be reentered is if the UR itself calls execute(). I don't think there's any way for a malicious token for instance, to get the UR to do this without breaking this lock, must be signed off by the original caller.